build/commands/lib/test.js

```diff
@@ -101,28 +32,35 @@ const test = async (
     Log.error('Running ios tests on a device is not yet supported')
     process.exit(1)
   }
-  await buildTests(suite, Config, options)
-  runTests(passthroughArgs, suite, Config, options)
-}
 
-const buildTests = async (suite, config, options = {}) => {
-  util.touchOverriddenFiles()
-  util.touchGsutilChangeLogFile()
+  const testsToRun = passthroughArgs.since
+    ? await getAffectedTests({ ...passthroughArgs, suite })
+    : getTestsToRun(Config, suite)
 
-  await util.buildTargets([suite], config.defaultOptions)
+  await buildTests(testsToRun, Config, options)
+  await runTests(passthroughArgs, { suite, testsToRun }, Config, options)
 }
 
-const deleteFile = (filePath) => {
-  if (fs.existsSync(filePath)) {
-    fs.unlinkSync(filePath)
+const buildTests = async (testsToRun, config) => {
+  config.buildTargets = testsToRun
+  if (testsToRun.length > 0) {
```

Comment:
I think we need to split up this PR because it's very hard to tell what has actually changed here. Can you create one PR that just splits up test.js into testUtils.js without making any other changes? This particular change also looks like an optimization unrelated to this PR so that should be separated out as well.

Reply:
It's required to avoid cyclic imports. It's not allowed in node 20 and up